src/components/VideoEditor.tsx

```diff
@@ -0,0 +1,229 @@
+import MenuItem from '@mui/material/MenuItem';
+import Stack from '@mui/material/Stack';
+import Alert from '@mui/material/Alert';
+import AlertTitle from '@mui/material/AlertTitle';
+import { Channel } from '../config/ambiance/channels';
+
+
+type VideoEditorProps = {
+  documentId: string,
+  video?: Partial<Ambiance>,
+  videoUrlCode?: string,
+  editVideo?: (documentId: string, videoUrlCode: string, newData: Ambiance) => void,
+  deleteVideo?: (documentId: string, videoUrlCode: string) => void,
+  addVideo?: (documentId: string, newData: Ambiance) => void
+}
+
+
+export default function VideoEditor(props: VideoEditorProps) {
+  const { video, documentId, videoUrlCode, editVideo, deleteVideo, addVideo } = props;
+  const [showDeleteAlert, setShowDeleteAlert] = useState<Boolean>(false);
+  const [showEditAlert, setShowEditAlert] = useState<Boolean>(false);
+  const [vidData, setVidData] = useState<Partial<Ambiance> | undefined>(video);
+
+  const verifyDelete = () => {
+    setShowDeleteAlert(true);
+  }
+
+  const verifyEdit = () => {
+    setShowEditAlert(true);
+  }
+
+  const deleteAction = () => {
+    if (!videoUrlCode || !deleteVideo) {
+      return;
+    }
+
+    deleteVideo(documentId, videoUrlCode);
+    setShowDeleteAlert(false);
+  }
+
+  const editVideoAction = () => {
+    if (!vidData || !videoUrlCode || !editVideo) {
+      return;
+    }
+
+    if (!vidData.name || !vidData.code || !vidData.group) {
+      alert("Name, Code, and Group are required");
+      return;
+    }
+
+    editVideo(documentId, videoUrlCode, vidData as Ambiance);
+    setShowEditAlert(false);
+  }
+
+  const addVideoAction = () => {
+    if (!vidData || !addVideo) {
+      return;
+    }
+
+    if (!vidData.name || !vidData.code || !vidData.group) {
+      alert("Name, Code, and Group are required");
+      return;
+    }
+
+    addVideo(documentId, vidData as Ambiance);
+    setVidData({});
+  }
+
+  const handleOnChange = (event: React.ChangeEvent<HTMLInputElement | HTMLTextAreaElement>, key: keyof Ambiance) => {
+    const newVidData: Partial<Ambiance> = {
+      ...vidData,
+      [key]: event.target.value
+    }
+    setVidData(newVidData);
+  }
+
+  const handleOnChangeChannel = (event: React.ChangeEvent<HTMLInputElement | HTMLTextAreaElement>, key: keyof Channel) => {
+    let newChannel: Channel;
+    if (key === "name") {
+      newChannel = {
+        link: vidData?.channel?.link || "",
+        name: event.currentTarget.value
+      }
+    } else {
+      newChannel = {
+        link: event.currentTarget.value,
+        name: vidData?.channel?.name || "",
+      }
+    }
+
+    const newVidData: Partial<Ambiance> = {
+      ...vidData,
+      channel: newChannel
+    }
+    setVidData(newVidData);
+  }
+
+
+  return (
+    <div id='Video-Editor'>
+      <Stack className='text-field-stack' direction="column" spacing={.4}>
+        <TextField
+          sx={{ '& .dsoQSg': { paddingTop: '10px' } }}
```

Comment:
Code Smell: having all of these have the exact same `sx={{ '& .dsoQSg': { paddingTop: '10px' } }}` prop. Can we either pull this out to a variable or apply the styling in a better way so it will be applied to all?